pkg/kubectl/cmd/auth/ican.go

```diff
@@ -0,0 +1,274 @@
+	return cmd
+}
+
+func (o *ICanOptions) Complete(f cmdutil.Factory, args []string) error {
+	if len(args) != 0 {
+		return errors.New("you should specify no arguments")
+	}
+	var err error
+	client, err := f.ClientSet()
+	if err != nil {
+		return err
+	}
+	o.SelfSARClient = client.Authorization()
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+
+	return err
+}
+
+func (o *ICanOptions) Validate() error {
+	if o.NonResourceOnly && o.ResourceOnly {
+		return errors.New("you should not specify --only-resources and --only-non-resource-urls at same time")
+	}
+
+	return nil
+}
+
+func (o *ICanOptions) RunAccessCheck() error {
+	var sar *authorizationapi.SelfSubjectRulesReview
+	sar = &authorizationapi.SelfSubjectRulesReview{
+		Spec: authorizationapi.SelfSubjectRulesReviewSpec{
+			Namespace: o.Namespace,
+		},
+	}
+	response, err := o.SelfSARClient.SelfSubjectRulesReviews().Create(sar)
+	if err != nil {
+		return err
+	}
+
+	err = o.printStatus(response.Status)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *ICanOptions) printStatus(status authorizationapi.SubjectRulesReviewStatus) error {
+	if status.Incomplete && !o.Quiet {
+		fmt.Fprintln(o.Out, "warning: the list may be incomplete")
+	}
+	if len(status.EvaluationError) != 0 && !o.Quiet {
+		fmt.Fprintln(o.Out, status.EvaluationError)
+	}
+	rules := []rbacv1.PolicyRule{}
+	if o.NonResourceOnly || o.ResourceOnly {
+		if o.NonResourceOnly {
+			rules = append(rules, convertNonResourceToPolicyRule(status)...)
+		}
+		if o.ResourceOnly {
+			rules = append(rules, convertResourceToPolicyRule(status)...)
+		}
+	} else {
+		//since we have validated that they can't be true at same time
+		//this block only handles printing all
+		rules = append(rules, convertNonResourceToPolicyRule(status)...)
+		rules = append(rules, convertResourceToPolicyRule(status)...)
+	}
+
+	breakdownRules := []rbacv1.PolicyRule{}
+	for _, rule := range rules {
+		breakdownRules = append(breakdownRules, validation.BreakdownRule(rule)...)
+	}
+
+	compactRules, err := validation.CompactRules(breakdownRules)
+	if err != nil {
+		return err
+	}
+	sort.Stable(rbacv1helpers.SortableRuleSlice(compactRules))
+
+	w := printers.GetNewTabWriter(o.Out)
+	defer w.Flush()
+
+	if o.NonResourceOnly {
+		if err := printNonResourceURLs(w, compactRules, o.NoHeader); err != nil {
+			return err
+		}
+		return nil
+	}
+	if o.ResourceOnly {
+		if err := printResources(w, compactRules, o.NoHeader); err != nil {
+			return err
+		}
+		return nil
+	}
+	if err := printAll(w, compactRules, o.NoHeader); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func printAll(out io.Writer, rules []rbacv1.PolicyRule, noHeader bool) error {
+	if !noHeader {
+		if _, err := fmt.Fprintf(out, "Resources\tNon-Resource URLs\tResource Names\tVerbs\n"); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintf(out, "---------\t-----------------\t--------------\t-----\n"); err != nil {
```

Comment:
`kubectl api-resources` doesn't have such line.

Reply:
@CaoShuFeng Since the output here is similar with RBAC clusterrole/role,  maybe we can just keep same with them.
And the output of `kubectl describe role xxxx` is like:
```
  Resources                                       Non-Resource URLs  Resource Names  Verbs
  ---------                                       -----------------  --------------  -----
  serviceaccounts                                 []                 []              [create delete deletecollection get list patch update watch impersonate]
  configmaps                                      []                 []              [create delete deletecollection get list patch update watch]
  endpoints                                       []                 []              [create delete deletecollection get list patch update watch]
  persistentvolumeclaims                          []                 []              [create delete deletecollection get list patch update watch]
  pods/attach                                     []                 []              [create delete deletecollection get list patch update watch]
  pods/exec                                       []                 []              [create delete deletecollection get list patch update watch]
  pods/portforward                                []                 []              [create delete deletecollection get list patch update watch]
  pods/proxy                                      []                 []              [create delete deletecollection get list patch update watch]
  pods                                            []                 []              [create delete deletecollection get list patch update watch]
  replicationcontrollers/scale                    []                 []              [create delete deletecollection get list patch update watch]
```